src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp

```diff
@@ -666,3 +692,72 @@ void runtime::intelgpu::do_logic_kernel(cldnn::topology& topology,
                                                  {1});
     topology.add(op_logical);
 }
+
+void runtime::intelgpu::do_reverse_operation(cldnn::topology& topology,
+                                             const string& input_name,
+                                             const Shape& input_shape,
+                                             const string& output_name,
+                                             const Shape& output_shape,
+                                             const element::Type& output_type,
+                                             const AxisSet& reversed_axes)
+{
+    const cldnn::layout layout = IntelGPULayout::create_cldnn_layout(output_type, output_shape);
+    const string entry_point_name = "reverse_" + output_name;
+    codegen::CodeWriter writer;
+    std::vector<size_t> gws;
+
+    writer << "__kernel void " << entry_point_name << "(const __global float input"
+           << array_dims(input_shape) << ", __global float output" << array_dims(output_shape)
+           << ")\n";
+
+    writer.block_begin();
+    {
+        size_t var_idx = 0;
+        // Current clDNN implementation supports only 3 dimension of global work sizes
+        for (auto const& i : output_shape)
+        {
+            if (var_idx < 3)
+            {
+                writer << "const unsigned i" << var_idx << " = get_global_id(" << var_idx << ");\n";
+                gws.push_back(i);
+                ++var_idx;
+            }
+        }
+        // Insert usual loops in case we have more than 3 dimension tensor
+        var_idx = 0;
+        for (auto const& i : output_shape)
+        {
+            if (var_idx > 2)
+            {
+                writer << "for (uint i" << var_idx << " = 0; i" << var_idx << " < " << i << "; ++i"
+                       << var_idx << ")\n";
+                writer.block_begin();
+            }
+            ++var_idx;
+        }
+        writer << "output" << access_dims(output_shape) << " = input"
+               << reverse_dims(output_shape, reversed_axes) << ";\n";
+
+        // Closing brackets for extra loops
+        var_idx = 0;
+        for (auto const& i : output_shape)
```

Comment:
The same as above. Please create `static vector<size_t> generate_loops_close()` for this.